### DISCUSSION TITLE
Prevent seeding production branch?

### DISCUSSION THREAD

**Discussion by thenbe** [planetscale/discussion#203] [Q&A] [answered]
My dev workflow involves connecting to one of the existing dev branches through the `pscale` cli and seeding it with a nodejs script. Currently, choosing the production branch is one arrow key click away from a dev branch when connecting through `pscale connect mydb`. I'm concerned that one day I, or a team member, will the run seed script after erroneously connecting to the production branch.

Is there a good method/workflow you would recommend that would decrease the chances of that happening?

**Comment by MyNameIsOka** [comment]
I guess you would need to check in your nodejs script if it's okay to execute that script in the database you are connected to.
Not sure how, though.

**Comment by jgreet** [comment, marked as answer]
@ambiguous48 on production branches you can't create or drop tables directly, you'll get an error message like this:

```sql
mysql> create table t;
ERROR 1105 (HY000): direct DDL is disabled
```

So if your seed script script drops or creates tables early enough it may already exit safely on that error. You could also try creating and dropping a table in the script to verify that you aren't on a production branch.

**Reply by thenbe** [reply]
This does the job. Thank you.

Btw will creating/dropping tables always result in an error if the branch is a production branch? Are there any settings we could possibly change that may affect this behavior?

**Reply by jgreet** [reply]
Yes, any DDL statement should fail on a production branch. There's no setting that will change this behavior.
